usr/src/uts/common/fs/zfs/spa.c

```diff
@@ -5471,6 +5472,7 @@ spa_vdev_remove(spa_t *spa, uint64_t guid, boolean_t unspare)
 		    ZPOOL_CONFIG_L2CACHE, l2cache, nl2cache, nv);
 		spa_load_l2cache(spa);
 		spa->spa_l2cache.sav_sync = B_TRUE;
+		spa_event_notify(spa, vd, ESC_ZFS_VDEV_REMOVE_AUX);
```

Comment:
@jeffpc it will always be NULL for the l2cache devices so yes you could explicitly pass `NULL` instead of `vd`. I'm assuming that the author only cared about the event an not the vdev payload.